homeassistant/components/xiaomi_miio/fan.py

```diff
@@ -434,35 +434,33 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
         if DATA_KEY not in hass.data:
             hass.data[DATA_KEY] = {}
 
-        host = config_entry.data[CONF_HOST]
-        token = config_entry.data[CONF_TOKEN]
         name = config_entry.title
         model = config_entry.data[CONF_MODEL]
         unique_id = config_entry.unique_id
-
-        _LOGGER.debug("Initializing with host %s (token %s...)", host, token[:5])
+        coordinator = hass.data[DOMAIN][config_entry.entry_id][KEY_COORDINATOR]
 
         if model in MODELS_PURIFIER_MIOT:
-            air_purifier = AirPurifierMiot(host, token)
+            air_purifier = hass.data[DOMAIN][config_entry.entry_id][KEY_DEVICE]
```

Comment:
The device is always stored under the same key in hass.data. We can move this out of the if/else block and use a variable named `device`.